tests/test_integration_scenarios.py

```
@@ -0,0 +1,512 @@
+    def stats_system(self):
+        """Create system with stats tracking."""
+        from gengowatcher.stats import StatsManager
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            stats_path = pathlib.Path(tmpdir) / "stats.json"
+            manager = StatsManager(stats_path=stats_path)
+            yield manager
+
+    def test_hourly_aggregation(self, stats_system):
+        """Test hourly statistics aggregation."""
+        # Deterministic distribution for peak‑hour calculation
```

Ruff (0.14.14)

Comment contains ambiguous `‑` (NON-BREAKING HYPHEN). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)